mlflow/projects/_project_spec.py

```diff
@@ -42,6 +42,39 @@ def load_project(directory):
 
     project_name = yaml_obj.get("name")
 
+    databricks_spark_job_yaml = yaml_obj.get("databricks_spark_job")
+    if databricks_spark_job_yaml is not None:
+        if "python_file" not in databricks_spark_job_yaml:
+            raise MlflowException("'python_file' field is required for databricks spark job.")
+
+        if "entry_points" in yaml_obj:
+            raise MlflowException("databricks spark job does not support setting 'entry_points'.")
+
+        if env_type.DOCKER in yaml_obj:
+            raise MlflowException(
+                "databricks spark job does not support setting docker environment."
+            )
+
+        if env_type.PYTHON in yaml_obj:
```

Comment:
Is this a breaking change?

Reply:
No.

all these checkings are only triggered when `if databricks_spark_job_yaml is not None` condition is satisfied.
i.e. only if user adds  `databricks_spark_job` section (new added feature) in the MLproject file, it triggers these checkings.